test/integration/task_polling_test.js

```diff
@@ -97,4 +98,43 @@ suite('Task Polling', () => {
       assert.equal(taskRes.run.reasonResolved, 'completed');
     });
   });
+
+  test('task polling at normal speed', async () => {
+    settings.billingCycleUptime(30);
+    settings.billingCycleInterval(40);
+
+    worker = new TestWorker(DockerWorker);
+    await worker.launch();
+    //make sure the polling timer is accurate
+    await base.testing.sleep(6000);
+
+    settings.billingCycleUptime(0);
+    await new Promise((accept, reject) => {
```

Comment:
So one of the things we've done in tests is using the wait_for_event module and you can do something like

let pollingMessage = await waitForEvent(worker, 'polling');
assert.equal(pollingMessage.message, 'polling interval adjusted', <some error message if this fails>)

Either this happens or the test will time out.